src/test/kotlin/me/spradling/gift/core/tests/unit/api/GetAccountsHandlerTests.kt

```diff
@@ -0,0 +1,99 @@
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DisplayName("When I call `handleRequest` on GetAccountsHandler,")
+internal class GetAccountsHandlerTests : UnitTestBase() {
+
+  private val getHandler = GetAccountsHandler(inMemoryStorageClient)
+
+  @Nested
+  @DisplayName("given no results")
+  inner class NoResults {
+
+    @Test
+    @DisplayName("request should fail with Not Found error")
+    fun failsWithNotFoundError() {
+      val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap()
+
+      verifyErrorResponse(response, ErrorDetails.RESOURCE_NOT_FOUND)
+    }
+  }
+
+  @Nested
+  @DisplayName("given a valid request")
+  inner class ValidRequest {
+
+    @BeforeEach
+    fun setup() {
+      inMemoryStorageClient.createAccount(storageConverter.convert(validAccounts["api"]!!))
+    }
+
+    @Nested
+    @DisplayName("given valid query parameters,")
+    inner class ValidQueryParameter {
+
+      @BeforeEach
+      fun setup() {
+        `when`(mockContext.queryParam("limit")).thenReturn(listOf("10")) // Return that we want 10 accounts when calling the "limit" query param
+      }
+
+      @Test
+      @DisplayName("should succeed with 200 code")
+      fun shouldReturnAccounts() {
+        val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap() // Execute request
+
+        assertThat(response.statusCode).isEqualTo(HttpResponseStatus.OK.code()) // Should have 200 code
+        assertThat(response.body.isPresent).isEqualTo(true)
+        val retrievedAccounts = response.body.get() as List<Account>
+        assertThat(listOf(validAccounts["api"]).containsAll(retrievedAccounts) &&
+            retrievedAccounts.containsAll(listOf(validAccounts["api"])))
+      }
+    }
+
+    @Nested
+    @DisplayName("given no account limit")
+    inner class NoLimitQuery {
+
+      @Test
+      @DisplayName("should pass with no parameters")
+      fun shouldPassWithNoQueryParams() {
+        val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap()
+
+        assertThat(response.statusCode).isEqualTo(HttpResponseStatus.OK.code())
+        assertThat(response.body.isPresent).isEqualTo(true)
+        val retrievedAccounts = response.body.get() as List<Account>
+        assertThat(listOf(validAccounts["api"]).size == retrievedAccounts.size &&
+            listOf(validAccounts["api"]).containsAll(retrievedAccounts) &&
+            retrievedAccounts.containsAll(listOf(validAccounts["api"])))
+      }
+
+    }
+
+    @Nested
+    @DisplayName("given incomplete account limit")
+    inner class WrongLimitQuery {
+
+      @Test
+      @DisplayName("should fail with wrong parameters")
+      fun shouldFailWithWrongQueryParams() {
+
+        `when`(mockContext.queryParam("limit")).thenReturn(listOf("a"))
+
+        val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap()
+
+        assertThat(response.statusCode).isEqualTo(HttpResponseStatus.UNPROCESSABLE_ENTITY.code())
+      }
+    }
+  }
+}
```

Comment:
i would make sure that we have an least one test for multiple entries. mocking out the db to return more than one and assert that you get all those entries. then it would be awesome to have a test to verify that the `LIMIT` functionality is working as expected.